ci/rust-version.sh

```diff
@@ -18,7 +18,9 @@
 if [[ -n $RUST_STABLE_VERSION ]]; then
   stable_version="$RUST_STABLE_VERSION"
 else
-  stable_version=1.65.0
+  # read rust version from rust-toolchain.toml file
+  source "./scripts/read-cargo-variable.sh"
+  stable_version=$(readCargoVariable channel "./rust-toolchain.toml")
```

Comment:
Does this work from any other directory? I had to spend awhile learning about `BASH_SOURCE` to get this to work to properly on the SPL repo :sweat: 

For example, if I'm in `runtime`, can I do `../cargo-nightly fmt`?